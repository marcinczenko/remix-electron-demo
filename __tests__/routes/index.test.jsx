import { vi } from 'vitest'
import { render, screen } from '@testing-library/react'

import Home from '../../app/routes/index'

vi.mock('@remix-run/react', async () => {
  const mod = await vi.importActual < typeof import('@remix-run/react') > ('some-path')
  return {
    ...mod,
    useLoaderData: vi.fn().mockReturnValue({
      userDataPath: 'test-path'
    })
  }
})

describe('index', () => {
  it('renders the app', () => {
    render(<Home />)

    screen.getByText('test-path')
  })

  it('looks great', () => {
    const { asFragment } = render(<Home />)

    expect(asFragment()).toMatchSnapshot('Index Route Snapshot')
  })
})
