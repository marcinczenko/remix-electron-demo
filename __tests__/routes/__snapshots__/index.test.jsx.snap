// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`index > looks great > Index Route Snapshot 1`] = `
<DocumentFragment>
  <main>
    <div
      class="w-screen h-screen flex flex-col justify-center items-center bg-blue-900 text-white"
    >
      <h1
        class="text-3xl font-bold"
      >
        remix-electron-demo
      </h1>
      <p
        class="text-xs"
      >
        test-path
      </p>
    </div>
  </main>
</DocumentFragment>
`;
